realm/realm-library/src/objectServer/java/io/realm/mongodb/sync/SyncConfiguration.java

```diff
@@ -461,14 +471,48 @@ public String getUrlPrefix() {
      * into Realms that can be synchronized independently.
      */
     public BsonValue getPartitionValue() {
+        checkPartitionConfiguration();
```

Comment:
Mostly because calling this on a flexible sync config is a logic error, there is `isPartionBasedSyncConfiguration()` if you are unsure. I'll add a throws.